opennlp-tools/src/main/java/opennlp/tools/util/DownloadUtil.java

```diff
@@ -148,6 +151,57 @@ public static <T extends BaseModel> T downloadModel(URL url, Class<T> type) thro
     }
   }
 
+  /**
+   * Validates the downloaded model.
+   *
+   * @param sha512          the url to get the sha512 hash
+   * @param downloadedModel the model file to check
+   * @throws IOException thrown if the checksum could not be computed
+   */
+  private static void validateModel(URL sha512, Path downloadedModel) throws IOException {
+    // Download SHA512 checksum file
+    String expectedChecksum;
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(sha512.openStream()))) {
+      expectedChecksum = reader.readLine();
+
+      if (expectedChecksum != null) {
+        expectedChecksum = expectedChecksum.split("\\s")[0].trim();
+      }
+    }
+
+    // Validate SHA512 checksum
+    final String actualChecksum = calculateSHA512(downloadedModel);
+    if (!actualChecksum.equalsIgnoreCase(expectedChecksum)) {
+      throw new IOException("SHA512 checksum validation failed. Expected: "
```

Comment:
Idea:
Could instead throw a more specific exception (type) here to indicate a validation step has failed. Might be helpful for the caller to decide if a retry if the download shall be attempted or not.